### PR TITLE
Refactor on_reagent_change to pass through the change type

### PR DESCRIPTION
**Pull request by optimumtact**
this is purely so the smoke circuit can reset the notification only when
reagents are actually added to it

### CODE REVIEW

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 1]
```diff
@@ -179,7 +179,7 @@
 /obj/item/mecha_parts/mecha_equipment/extinguisher/get_equip_info()
 	return "[..()] \[[src.reagents.total_volume]\]"
 
-/obj/item/mecha_parts/mecha_equipment/extinguisher/on_reagent_change()
+/obj/item/mecha_parts/mecha_equipment/extinguisher/on_reagent_change(var/changetype)
 	return
```
Comment (optimumtact) [inline code/game/mecha/equipment/tools/work_tools.dm]: wat

File: code/modules/mob/living/simple_animal/bot/secbot.dm [hunk 2]
```diff
@@ -54,7 +54,7 @@
 	new /obj/item/stock_parts/cell/potato(Tsec)
 	var/obj/item/reagent_containers/food/drinks/drinkingglass/shotglass/S = new(Tsec)
 	S.reagents.add_reagent("whiskey", 15)
-	S.on_reagent_change()
+	S.on_reagent_change(ADD_REAGENT)
```
Comment (optimumtact) [inline code/modules/mob/living/simple_animal/bot/secbot.dm]: wait this is stupid, add_reagent calls this

Comment (optimumtact) [reply code/modules/mob/living/simple_animal/bot/secbot.dm]: 🤔

File: code/datums/datumvars.dm [hunk 3]
```diff
@@ -31,7 +31,7 @@
 	.["Show VV To Player"] = "?_src_=vars;[HrefToken(TRUE)];expose=[REF(src)]"
 
 
-/datum/proc/on_reagent_change()
+/datum/proc/on_reagent_change(var/changetype)
```
Comment (ACCount12) [inline code/datums/datumvars.dm]: Do we use `var/` in proc args?

File: code/modules/mob/living/simple_animal/bot/secbot.dm (hunk 2 above)
Comment (ACCount12) [reply code/modules/mob/living/simple_animal/bot/secbot.dm]: Just remove it then.

**Comment by ShizCalev** [comment]
Conflict.

File: code/datums/datumvars.dm (hunk 3 above)
Comment (Cyberboss) [reply code/datums/datumvars.dm]: we don't

File: code/modules/mob/living/simple_animal/bot/secbot.dm (hunk 2 above)
Comment (optimumtact) [reply code/modules/mob/living/simple_animal/bot/secbot.dm]: nah can't be bothered to confirm it works after still

**Pull request merged**